GillianCore/utils/list_utils.ml

```diff
@@ -236,3 +236,9 @@ let get_and_remove_nth n l =
         else x :: aux (i + 1) r
   in
   (!found, aux 0 l)
+
+let map_head f = function
+  | [] -> []
+  | x :: xs -> f x :: xs
+
+let map_last f l = l |> List.rev |> map_head f |> List.rev
```

Comment:
```ocaml
let[@tail_mod_cons] map_last f l =
   match l with
   | [] -> []
   | [ a ] -> [ f a ]
   | x::r -> x :: map_last f l
```

Reply:
Resolved in aaa58c2889789854b009175b6fbd8d3bab648d09